aten/src/ATen/native/native_functions.yaml

```diff
@@ -6333,13 +6333,31 @@
     MPS: set_mps_
   autogen: set, set.out
 
+# Not making it CompositeImplicitAutograd because lift
+# should be a primitive w.r.t. functorch
+
+# TODO: this should have a view annotation
+# TODO: shouldn't be a method
 - func: lift(Tensor self) -> Tensor
   variants: method
   dispatch:
-    # Not making it CompositeImplicitAutograd because lift
-    # should be a primitive w.r.t. functorch
     CompositeExplicitAutograd: lift
 
+# lift_fresh is called with an argument that is guaranteed to be
+# fresh (i.e., newly allocated).  This is ONLY called from a
+# torch.tensor call; if you FX trace a lift_fresh, you are obligated
+# to convert this into a lift_fresh_copy (because FX will violate the
+# freshness invariant when tracing).
+- func: lift_fresh(Tensor(a) self) -> Tensor(a)
```

Comment:
(not blocking this PR) This is my fault because it was already wrong before, but I wonder if we'll run into trouble with the alias annotations. There were Reasons that we had to make `lift` a primitive op (`CompositeExplicitAutograd`), but that's still a bit iffy because the vanilla kernel returns a view, but the functionalization kernel doesn't.

Reply:
since `lift_fresh` is `CompositeExplicitAutograd`, it'll get a generated autograd kernel, which will have those debug asserts that check if the output storage is a view, since we've given the op view annotations.

But let's say that somehow, the autograd kernel redispatches, and hits the functionalization kernel for `lift_fresh`, which returns a fresh tensor with a fresh `StorageImpl`. I think we might trip the assert in that case? (I'm pretty sure this would happen if... you do a `grad(functionalize(`torch.tensor()`))` ?)

Reply:
whoops commented got deleted

Since `lift_fresh` is a `CompositeExplicitAutograd` kernel and has alias annotations, we'll generate an autograd kernel that includes debug asserts that the output tensor aliases storage with the input.

I wonder if we'll run into trouble if we arrange for the autograd kernel to redispatch and hit the functionalization kernel for `lift_fresh`, which will generate a new `FunctionalTensorWrapper` with its own fresh storage. I think you can arrange for this to happen with something like `functionalize(grad(torch.tensor(...)))`?

Reply:
oh debug asserts. Sigh, I'm not sure. Will have to think about it